### PR TITLE
Change methods to const in InverseKinematics class

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/idyntree/issues/823 .

### CODE REVIEW

**Comment by traversaro** [comment]
I fixed the MATLAB bindings manually because I don't want to have the failing CI, and I want to regenerated them once I merged all the removal related to deprecations. The CI is now ready for review.

**Comment by traversaro** [comment]
The tests are failing, however I tried to just take the changes from https://github.com/robotology/idyntree/issues/823 and apply them on devel. @GiulioRomualdi do you have any idea?

File: src/inverse-kinematics/src/InverseKinematics.cpp [hunk 1]
```diff
@@ -1398,8 +1396,7 @@ namespace iDynTree {
 #endif
     }
 
-    // this should be const reference but we keep it like this to avoid breaking the API. Please change me in iDynTree 4.0
-    void InverseKinematics::setCOMConstraintProjectionDirection(iDynTree::Vector3 direction)
+    void InverseKinematics::setCOMConstraintProjectionDirection(const iDynTree::Vector3 direction)
```
Comment (GiulioRomualdi) [inline src/inverse-kinematics/src/InverseKinematics.cpp:1399]:
```suggestion
    void InverseKinematics::setCOMConstraintProjectionDirection(const iDynTree::Vector3& direction)
```

File: src/inverse-kinematics/include/iDynTree/InverseKinematics.h [hunk 2]
```diff
@@ -1090,7 +1090,7 @@ class iDynTree::InverseKinematics
      *
      * @param direction    vector along which we want to project a point
      */
-    void setCOMConstraintProjectionDirection(iDynTree::Vector3 direction);
+    void setCOMConstraintProjectionDirection(const iDynTree::Vector3 direction);
```
Comment (GiulioRomualdi) [inline src/inverse-kinematics/include/iDynTree/InverseKinematics.h:1093]:
```suggestion
    void setCOMConstraintProjectionDirection(const iDynTree::Vector3& direction);
```

File: src/inverse-kinematics/src/InverseKinematics.cpp (hunk 1 above)
Comment (traversaro) [reply src/inverse-kinematics/src/InverseKinematics.cpp:1399]: Not that I am against it, but consider that for such as small class, I am not sure if the advantage of passing by reference in term of avoiding copy  (pro) is more then the disavantage of having an additional indirection.

**Comment by traversaro** [comment]
@GiulioRomualdi CI passed except for the slow macOS jobs, I think this is ready for the new review.

**Review by GiulioRomualdi — APPROVED** [review]

**Pull request merged**